### PR TITLE
Add unit tests for com.alibaba.jvm.sandbox.api.util.GaStringUtils

### PR DESCRIPTION
**Pull request by Braavos96**
I've analysed your codebase and noticed that `com.alibaba.jvm.sandbox.api.util.GaStringUtils` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

**Comment by airfer** [comment]
it helpful～

**Comment by Braavos96** [comment]
Hi @airfer just following up and wondering if you would like me to make changes to the PR?

**Comment by airfer** [comment]
> Hi @airfer just following up and wondering if you would like me to make changes to the PR?

The tests for GaStringUtils covers most situations, but the matching func needs more attention, especially about core features, eg 'nextIsNotWildcard'. `Assert.assertFalse(GaStringUtils.matching("foo", "\\o"))`  is a useless case and doesn't cover the intention of the feature. Besides, '?' & '*' needs more cases, eg in the middle position of the string.

File: sandbox-api/src/test/java/com/alibaba/jvm/sandbox/api/util/GaStringUtilsTests.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Test
+    public void testMatching() {
+        Assert.assertFalse(GaStringUtils.matching(null, "bar"));
+        Assert.assertFalse(GaStringUtils.matching("foo", null));
+        Assert.assertFalse(GaStringUtils.matching(null, null));
+        Assert.assertFalse(GaStringUtils.matching("foo", "bar"));
+        Assert.assertFalse(GaStringUtils.matching("foobar", "foo"));
+        Assert.assertFalse(GaStringUtils.matching("foobar", "*a"));
+        Assert.assertFalse(GaStringUtils.matching("foo", "\\o"));
```
Comment (airfer) [inline sandbox-api/src/test/java/com/alibaba/jvm/sandbox/api/util/GaStringUtilsTests.java]: //Verify nextIsNotWildcard  features  and add  more cases about '*' &  '?'
` Assert.assertFalse(GaStringUtils.matching("foo", "\\*"));
        Assert.assertFalse(GaStringUtils.matching("foo", "f\\?o"));
        Assert.assertFalse(GaStringUtils.matching("fooMatching", "fool\\*ing"));
      Assert.assertTrue(GaStringUtils.matching("foo", "f?o"));
 Assert.assertTrue(GaStringUtils.matching("fooMatching", "foo*ing"));
        Assert.assertTrue(GaStringUtils.matching("fooMatching", "foo*"));`

Comment (Braavos96) [reply sandbox-api/src/test/java/com/alibaba/jvm/sandbox/api/util/GaStringUtilsTests.java]: Test cases added. Thank you :)

**Comment by oldmanpushcart** [comment]
well done !

**Pull request merged**